blog/content/post/2020-04-03-trenchboot-nlnet-lz-validation.md

```diff
@@ -0,0 +1,796 @@
+`nixpkgs` with previously downloaded one. Make sure the directory is valid (we
+have it in home (~)). If you follow our instruction step-by-step, you have it
+also there.
+
+10. Reboot platform.
+
+DRTM is not enabled yet! Boot to NixOS and finish configuration.
+
+11. Install GRUB2-TrenchBoot to `/dev/sdX`.
+
+    ```bash
+    $ grub-install /dev/sda
+    ```
+
+Remember to choose proper device (disk) - in our case it is `/dev/sda`.
+
+12. Ensure that `slaunch` modules are present in `/boot/grub/i386-pc/`.
+
+    ```bash
+    $ ls /boot/grub/i386-pc | grep slaunch
+    slaunch.mod
+    slaunch.module
+    ```
+
+13. Find Landing Zone package in `/nixos/store/`.
+
+    ```bash
+    $ ls /nix/store/ |grep landing-zone
+    62ik61qxadavc2xix4sm8mbm0fcxlz2i-landing-zone-1.0
+    ```
+
+14. Copy `lz_header.bin` to `/boot/` directory.
+
+    ```bash
+    $ cp /nix/store/62ik61qxadavc2xix4sm8mbm0fcxlz2i-landing-zone-1.0/lz_header.bin /boot/lz_header
+    ```
+
+15. Check `/boot/grub/grub.cfg` file and its `NixOS - Default` menu entry.
+Adjust `/etc/nixos/configuration.nix` and its `boot.loader.grub.extraEntries`
+line to have exactly the same directories included.
+
+    ```bash
+    $ cat /boot/grub/grub.cfg
+    (...)
+    menuentry "NixOS - Default" {
+    search --set=drive1 --fs-uuid 178473b0-282f-4994-96fc-a8e51e2cfdac
+    search --set=drive2 --fs-uuid 178473b0-282f-4994-96fc-a8e51e2cfdac
+      linux ($drive2)/nix/store/ymvcgas7b1bv76n35r19g4p142v4cr0b-linux-5.1.0/bzImage systemConfig=/nix/store/b32wgz392q99cls12pkd8adddzbdkprn-nixos-system-nixos-20.09.git.50c3e448fceM init=/nix/store/b32wgz392q99cls12pkd8adddzbdkprn-nixos-system-nixos-20.09.git.50c3e448fceM/init console=ttyS0,115200 earlyprintk=serial,ttyS0,115200 loglevel=4
+      initrd ($drive2)/nix/store/zv2vl35xldkbss1y2fib1nifmw0yvick-initrd-linux-5.1.0/initrd
+    }
+    (...)
+    ```
+
+With `grub.cfg` content as above `configuration.nix` must have
+`boot.loader.grub.extraEntries `line like this:
+
+    ```bash
+    $ cat /etc/nixos/configuration.nix
+      (...)
+      boot.loader.grub.extraEntries = ''
+      menuentry "NixOS - Secure Launch" {
+      search --set=drive1 --fs-uuid 178473b0-282f-4994-96fc-a8e51e2cfdac
+      search --set=drive2 --fs-uuid 178473b0-282f-4994-96fc-a8e51e2cfdac
+        slaunch skinit
+        slaunch_module ($drive2)/boot/lz_header
+        linux ($drive2)/nix/store/ymvcgas7b1bv76n35r19g4p142v4cr0b-linux-5.1.0/bzImage systemConfig=/nix/store/b32wgz392q99cls12pkd8adddzbdkprn-nixos-system-nixos-20.09.git.50c3e448fceM init=/nix/store/b32wgz392q99cls12pkd8adddzbdkprn-nixos-system-nixos-20.09.git.50c3e448fceM/init console=ttyS0,115200 earlyprintk=serial,ttyS0,115200 loglevel=4
+        initrd ($drive2)/nix/store/zv2vl35xldkbss1y2fib1nifmw0yvick-initrd-linux-5.1.0/initrd
+      }
+    '';
+    ```
+
+If there are differences in any of `search --set=drive1...`, `search
+--set=drive2...`, `linux ($drive2)/nix/store...` lines. Edit
```

Comment:
```suggestion
--set=drive2...`, `linux ($drive2)/nix/store...` lines, edit
```

Reply:
@piotr-kleins I guess at least hashes may be different (not sure how they are generated, though). I think this must be empathized so users won't end up with non-working configuration.